### PR TITLE
Convert to typescript

### LINKED ISSUE #76: Convert to Typescript

### PR DESCRIPTION
**Pull request by Devon7925**
fixes #76 

Problems to address:
- [x] Dependencies are missing types
- [x] Slang WASM module doesn't have types. Probably retrievable during slang compile?
- [x] Almost every `document.getElementById` creates an error
- [x] `canvas`, `renderOutput`, `renderOverlay`, `targetResultContainer`, `tooltip` at the global scope doesn't seem to actually be defined/assigned anywhere, I'd like more information before fixing this
- [x] `monaco.MarkerSeverity.Information` doesn't seem to actually exist but used in `language-server.ts`?
- [x] `createBindGroup` expects 0 arguments but is passed 1 in `pipeline.createBindGroup(allocatedResources)` in `try-slang.ts`
- [x] `setLanguage` does not exist on `ITextModel` in `try-slang.ts`
- [x] Make sure everything still works
- [x] Look into jjsontree.js title bar modification(https://github.com/shader-slang/slang-playground/pull/77#discussion_r1872609691)
- [x] Make sure ci works
- [x] Build instructions need rewriting

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -9,14 +9,9 @@
     content="Interactive playground for experimenting with Slang shading language. Write, compile, and visualize shader code in real-time." />
   <meta name="keywords" content="Slang, shader, WebGPU, graphics programming, code playground, real-time rendering" />
   <link href="./styles/styles.css" rel="stylesheet" />
-  <link href="json-tree/jsontree.js.css" rel="stylesheet" />
+  <link rel="stylesheet" href="node_modules/jjsontree.js/dist/jsontree.js.css">
```
Comment (csyonghe) [inline index.html:12]: I believe we modified the jaontree css file a bit to hide the title bar.

**Review by csyonghe — COMMENTED** [review]
I am not seeing the CI command to compile the ts, is this automatic?

File: index.html (hunk 1 above)
Comment (Devon7925) [reply index.html:12]: ah, ok. Will look into it once I get it working.

**Comment by Devon7925** [comment]
Right now, I'm just trying to fix all the type errors, after that I'll make sure the website works properly, and then after that make sure ci works properly. The changes I've currently made to ci were mostly to get a wasm build of slang-tools. I'm not actually sure why slang-tools is used and it doesn't seem necessary for the site to work, but I'm trying to make minimal changes when possible.

**Comment by Devon7925** [comment]
aka right now I haven't even compiled to js locally

**Comment by Devon7925** [comment]
I have now verified the actions script works. See deploy at https://devon7925.github.io/slang-playground/

**Comment by csyonghe** [comment]
The deployment has some regressions on the bugs we recently fixed, so it needs to be merged with some recent commits.

**Comment by csyonghe** [comment]
See changes in https://github.com/shader-slang/slang-playground/pull/74/files

**Comment by Devon7925** [comment]
> The deployment has some regressions on the bugs we recently fixed, so it needs to be merged with some recent commits.

Apologies, I forgot to rerun deploy after merge. Redeployed so it should be fixed now.

**Comment by csyonghe** [comment]
This is fantastic. Thank you for the hard work @Devon7925! This will make the code much easier to maintain going forward.

**Pull request merged**